backend/src/bork-protocol/utils/selection/select-account.ts

```diff
@@ -0,0 +1,226 @@
+import { accountTopicQueries, tweetQueries } from '@/extensions/src/db/queries';
+import type { TargetAccount, WeightedAccount } from '@/types/account';
+import type { TwitterConfig } from '@/types/config';
+import { type IAgentRuntime, elizaLogger } from '@elizaos/core';
+import { updateYapsData } from '../account-metrics/yaps';
+import { getAggregatedTopicWeights } from '../topic-weights/topics';
+import { analyzeTopicRelationships } from './analyze-topic-relationships';
+
+/**
+ * Performs weighted random selection of accounts from a pool
+ * @param accounts Array of weighted accounts to select from
+ * @param count Number of accounts to select
+ * @returns Array of selected target accounts
+ */
+function selectAccountsWithWeights(
+  accounts: WeightedAccount[],
+  count: number,
+): TargetAccount[] {
+  const selected: TargetAccount[] = [];
+  const available = [...accounts];
+  const totalWeight = accounts.reduce((sum, { weight }) => sum + weight, 0);
+
+  for (let i = 0; i < count && available.length > 0; i++) {
+    let randomWeight = Math.random() * totalWeight;
+    let selectedIndex = 0;
+
+    // Find the account that corresponds to the random weight
+    for (let j = 0; j < available.length; j++) {
+      randomWeight -= available[j].weight;
```

Comment:
Probably need to randomize available array. If the first item is very big in weight it will have a good chance of being picked.